h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/matrix/MatrixFactoryUtils.java

```diff
@@ -16,14 +16,14 @@ public static int setResponseAndWeight(Chunk[] chunks, int respIdx, int weightId
         return j;
     }
 
-    public static int setResponseAndWeight(Vec.Reader w, float[] resp, float[] weights, Vec.Reader respVec, int j, long i) {
-        if (w != null) {
-            if(w.at(i) == 0) {
+    public static int setResponseAndWeight(Chunk weightChunk, Chunk respChunk, float[] resp, float[] weights, int j, int i) {
```

Comment:
Using the chunks directly seems like good improvement to me. No need to create readers.

Just thinking out loud: Could this method's logic be inlined somehow ? It is invoked once per line (we have big tests with hundreds of gigabytes and tens of millions of lines). A guarantee of saving us from copying 4 pointers & 32 + 64bit number once per line could mean some performance gain ? I'd investigate how JVM behaves to this method.